packages/client/components/EditableText.tsx

```diff
@@ -87,62 +87,34 @@ interface Props {
   onEditingChange?: (isEditing: boolean) => void
 }
 
-interface State {
-  autoFocus: boolean
-  value: string
-  isEditing: boolean
-}
-
-class EditableText extends Component<Props, State> {
-  static getDerivedStateFromProps(
-    nextProps: Readonly<Props>,
-    prevState: State
-  ): Partial<State> | null {
-    const {initialValue} = nextProps
-    if (prevState.isEditing || initialValue === prevState.value) return null
-    return {value: initialValue}
-  }
-
-  state = {
-    autoFocus: false,
-    isEditing: false,
-    value: this.props.initialValue
+const EditableText = forwardRef((props: Props, ref: any) => {
```

Comment:
@tianrunhe @tiffanyhan @nickoferrall 
in the past, I know we've talked about functional vs. class components & how we're almost entirely using functional components now. This is probably the best example we'll have of converting a legacy `Class` component to a functional component-- thought it'd be fun to compare side to side (~50 LOCs shorter with hooks)